### PR TITLE
Add Customisable Maximum Token Lifetime

### PR DESCRIPTION
**Pull request by artskidesign**
This PR updates the token.php file to allow external plugins to set a custom maximum token lifetime.

**Changes include**:
- New Filter Added: Introduced the vip_decoupled_max_token_lifetime filter.
- Updated Function: Modified get_token_lifetime_in_seconds to use this filter, allowing the maximum token lifetime ($max_lifetime) to be adjusted externally.

**Impact**:
Allows more flexibility in setting token lifetimes through external plugins.
Maintains default maximum lifetime unless specifically overridden.

### CODE REVIEW

**Comment by smithjw1** [comment]
@chriszarate I'm interested in your take on this from a security standpoint. Longer-lived tokens are riskier. Should we set a "supermax" lifetime or leave this completely configurable by bundle users?

**Comment by pwestern** [comment]
@smithjw1 just to quickly weigh in here - we have a business case to extend preview token lifetimes. In our case, we don't have any security concerns, but appreciate this won't apply to everyone. Having a way to extend the lifetime beyond 3hrs would be super helpful, so any hooks to configure would be really appreciated

**Comment by chriszarate** [comment]
> Longer-lived tokens are riskier. Should we set a "supermax" lifetime or leave this completely configurable by bundle users?

I'm open to increasing the default max. Three hours was a somewhat arbitrary decision. The risk of a long-lived token is scoped to "viewing the latest revision of a specific post." I think different organizations will have different ideas about how to interpret that risk.

@pwestern @artskidesign What would a more reasonable max token lifetime look like for your use case?

**Comment by pwestern** [comment]
@chriszarate the request we've had from the business is to extend the lifetime indefinitely - which we appreciate is far longer than many other users are likely to need. Our hope was that there would be a way to override the default max token lifetime so we could set it to 1 year.

File: preview/token.php [hunk 1]
```diff
@@ -25,7 +25,20 @@ function get_meta_key() {
...
 	$default_lifetime = HOUR_IN_SECONDS;
-	$max_lifetime     = 3 * HOUR_IN_SECONDS;
+	$default_max_lifetime = 3 * HOUR_IN_SECONDS;
+
+	/**
+	 * Filter the maximum allowed token lifetime.
+	 *
+	 * This filter allows external plugins or custom code to modify the maximum
+	 * lifetime of a token. It provides a way to enforce an upper limit on the 
+	 * duration for which a token remains valid.
```
Comment (smithjw1) [inline preview/token.php:35]: Would you consider adding language here to match other language in the plugin:

Understand the security implications of this change: the longer the token lifetime, the more security risk you bear with a compromised token. Extend the maximum lifetime with caution.

Comment (artskidesign) [reply preview/token.php:35]: @smithjw1 confirming additional comment as been added as per your suggestion.

Comment (artskidesign) [reply preview/token.php:35]: @smithjw1 @chriszarate let us know if there is anything we should change here and when would you be able to push things to production. Thanks again for your assistance.

**Review by smithjw1 — APPROVED** [review]
Thanks for making this suggestion, happy to approve these changes.

**Comment by alecgeatches** [comment]
Merging these changes in. Thank you @artskidesign!

**Pull request merged**